src/orchestrator/unary.rs

```diff
@@ -634,7 +679,7 @@ pub async fn detect_for_generation(
     debug!(%detector_id, ?request, "sending generation detector request");
     let response = ctx
         .detector_client
-        .text_generation(&detector_id, request)
+        .text_generation(&detector_id, request, headers.clone())
```

Comment:
Drop unnecessary clone

```suggestion
        .text_generation(&detector_id, request, headers)
```